src/main/java/org/gridsuite/study/server/StudyService.java

```diff
@@ -359,385 +366,447 @@ public StudyEntity doGetStudyWithPreFetchedLoadFlowResultAndUpdateIsPrivate(UUID
-                    multipartBodyBuilder.part("file", file);
-
-                    return webClient.post()
-                            .uri(caseServerBaseUri + "/" + CASE_API_VERSION + "/cases/private")
-                            .header(HttpHeaders.CONTENT_TYPE, MediaType.MULTIPART_FORM_DATA.toString())
-                            .body(BodyInserters.fromMultipartData(multipartBodyBuilder.build()))
-                            .retrieve()
-                            .onStatus(httpStatus -> httpStatus != HttpStatus.OK, clientResponse ->
-                                    handleStudyCreationError(studyUuid, userId, isPrivate, clientResponse, "case-server")
-                            )
-                            .bodyToMono(UUID.class)
-                            .publishOn(Schedulers.boundedElastic())
-                            .log(ROOT_CATEGORY_REACTOR, Level.FINE);
-                })
-                .doOnError(t -> !(t instanceof StudyException), t -> emitStudyCreationError(studyUuid, userId, isPrivate, t.getMessage()));
+            .flatMap(file -> {
+                MultipartBodyBuilder multipartBodyBuilder = new MultipartBodyBuilder();
+                multipartBodyBuilder.part("file", file);
+
+                return webClient.post()
+                    .uri(caseServerBaseUri + "/" + CASE_API_VERSION + "/cases/private")
+                    .header(HttpHeaders.CONTENT_TYPE, MediaType.MULTIPART_FORM_DATA.toString())
+                    .body(BodyInserters.fromMultipartData(multipartBodyBuilder.build()))
+                    .retrieve()
+                    .onStatus(httpStatus -> httpStatus != HttpStatus.OK, clientResponse ->
+                        handleStudyCreationError(studyUuid, userId, isPrivate, clientResponse, "case-server")
+                    )
+                    .bodyToMono(UUID.class)
+                    .publishOn(Schedulers.boundedElastic())
+                    .log(ROOT_CATEGORY_REACTOR, Level.FINE);
+            })
+            .doOnError(t -> !(t instanceof StudyException), t -> emitStudyCreationError(studyUuid, userId, isPrivate, t.getMessage()));
     }
 
-    Mono<byte[]> getVoltageLevelSvg(UUID networkUuid, String voltageLevelId, boolean useName, boolean centerLabel, boolean diagonalLabel,
-                                    boolean topologicalColoring, String componentLibrary) {
-        String path = UriComponentsBuilder.fromPath(DELIMITER + SINGLE_LINE_DIAGRAM_API_VERSION + "/svg/{networkUuid}/{voltageLevelId}")
-                .queryParam("useName", useName)
-                .queryParam("centerLabel", centerLabel)
-                .queryParam("diagonalLabel", diagonalLabel)
-                .queryParam("topologicalColoring", topologicalColoring)
-                .queryParamIfPresent("componentLibrary", Optional.ofNullable(componentLibrary))
+    Mono<byte[]> getVoltageLevelSvg(UUID studyUuid, String voltageLevelId, DiagramParameters diagramParameters, UUID nodeUuid) {
```

Comment:
Perhaps  a SLDService ?

Reply:
Maybe in another US : we could do the same also for calls to case-server, actions-server, conversion-server, loadflow-server, security-analysis-server, geo-data-server ...